server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java

```diff
@@ -85,18 +83,14 @@ protected boolean resolveIndex(MultiGetShardRequest request) {
 
     @Override
     protected ShardIterator shards(ClusterState state, InternalRequest request) {
-        ShardIterator shards = clusterService.operationRouting()
+        ShardIterator iterator = clusterService.operationRouting()
             .getShards(state, request.request().index(), request.request().shardId(), request.request().preference());
```

Comment:
Could it be null here as well?

Reply:
Yes. Good point. This is how our current workaround works. There should be a null check there. I'll adjust it with this PR.

Reply:
I opened https://github.com/elastic/elasticsearch/pull/94164 for this.